### PR TITLE
`operator+` in RipCurrent.

### PR DESCRIPTION
**Pull request by dkorolev**
A few shortcuts here and there, but should be good enough for the review. Thanks Max!

### CODE REVIEW

File: RipCurrent/ripcurrent.h [hunk 1]
```diff
@@ -289,60 +215,113 @@ class SharedDefinition<LHSTypes<LHS_TYPES...>, RHSTypes<RHS_TYPES...>> {
...
+// Implementation-wise, this implies only the first to initialize "plus combiner" should create an MMPQ, while all the
+// other plus-combiners should receive messages from the top-level one synchronously, with no MMPQs or mutexes involved.
+
 template <class LHS_TYPELIST, class RHS_TYPELIST>
 class SubCurrentScope;
 
 template <class... LHS_TYPES, class... RHS_TYPES>
-class SubCurrentScope<LHSTypes<LHS_TYPES...>, RHSTypes<RHS_TYPES...>> : public EmitDestination<LHSTypes<LHS_TYPES...>> {
+class SubCurrentScope<LHSTypes<LHS_TYPES...>, RHSTypes<RHS_TYPES...>>
+    //: public BlockOutgoingInterface<ThreadUnsafeOutgoingTypes<LHS_TYPES...>> {
```
Comment (mzhurovich) [inline RipCurrent/ripcurrent.h]: Left here intentionally?

File: RipCurrent/ripcurrent.h [hunk 2]
```diff
@@ -457,93 +441,101 @@ class EventConsumersManager final {
...
-
-template <class... LHS_TYPES, class... RHS_TYPES, typename USER_CLASS>
-class EventConsumerInitializer<LHSTypes<LHS_TYPES...>, RHSTypes<RHS_TYPES...>, USER_CLASS> final {
+// `UserClassInstantiator` instantiates the user class passed in as `USER_CLASS`.
+// It serves two purposes:
+// 1) Itself, it inherits from `BlockIncomingInterface<ThreadSafeIncomingTypes<LHS_TYPES...>>`, and can accept entries.
+//    Those entries are assumed thread safe, and are proxied directly to the user code's `.f()` method.
+// 2) It requires the `next_` parameter, which is a `BlockOutgoingInterface<ThreadUnsafeOutgoingTypes<RHS_TYPES...>>`.
+//    Prior to instantiating user class, it uses the `BlockCallsConsumersManager::CallsConsumerLifetimeScope` mechanism
+//    to enable use code to make the calls to `emit<>`, `post<>`, `schedule<>`, and `head<>` from its constructor.
```
Comment (mzhurovich) [inline RipCurrent/ripcurrent.h]: `use code` -> `user code`?

File: RipCurrent/ripcurrent.h [hunk 3]
```diff
@@ -816,6 +816,143 @@ SharedCurrent<LHS_TYPELIST, RHS_TYPELIST> operator|(SharedCurrent<LHS_TYPELIST,
...
+      }
+
+      void operator()(CurrentSuper&&) {
+        // Should define this method to make sure the RTTI call compiles.
+        CURRENT_ASSERT(false);
+      }
+
+     private:
+      Scope* self_;
+      movable_message_t x_;
```
Comment (mzhurovich) [inline RipCurrent/ripcurrent.h]: unused?

File: RipCurrent/ripcurrent.h (hunk 1 above)
Comment (dkorolev) [inline RipCurrent/ripcurrent.h]: :+1: 3b704f2

File: RipCurrent/ripcurrent.h (hunk 3 above)
Comment (dkorolev) [inline RipCurrent/ripcurrent.h]: :+1: 3b704f2

File: RipCurrent/types.h [hunk 4]
```diff
@@ -142,7 +147,7 @@ template <typename T>
 struct LHSTypesFromTypeListImpl;
 
 template <typename... TS>
-struct LHSTypesFromTypeListImpl<TypeListImpl<TS...>> {
+struct LHSTypesFromTypeListImpl<TypeListImpl<TS...> > {
```
Comment (mzhurovich) [inline RipCurrent/types.h]: Hmm.

Comment (dkorolev) [inline RipCurrent/types.h]: Oh well.

**Pull request merged**